deploy.py

```diff
@@ -68,27 +68,47 @@ def deploy():
     # proxy.secretTokens have leaked. So let's be careful with that!
     PROXY_SECRET_KEY = bytes.fromhex(os.environ['PROXY_SECRET_KEY'])
 
-    clusters = parse_clusters()
-    for cluster in clusters:
+    if cluster and hub:
         with cluster.auth():
-            for hub in cluster.hubs:
-                hub.deploy(k, PROXY_SECRET_KEY)
+            hub.deploy(k, PROXY_SECRET_KEY)
+    else:
+        clusters = parse_clusters()
+        for cluster in clusters:
+            with cluster.auth():
+                for hub in cluster.hubs:
+                    hub.deploy(k, PROXY_SECRET_KEY)
 
 
 def main():
     argparser = argparse.ArgumentParser()
-    argparser.add_argument(
-        'action',
-        choices=['build', 'deploy']
+    subparsers = argparser.add_subparsers(dest='action')
+
+    build_parser = subparsers.add_parser('build')
+    deploy_parser = subparsers.add_parser('deploy')
+
+    deploy_subparsers = deploy_parser.add_subparsers(dest='mode')
+
+    all_parser = deploy_subparsers.add_parser(
```

Comment:
How about the following commandline forms:

`python3 deploy.py deploy` deploys all clusters and hubs

`python3 deploy.py deploy <cluster-name>` deploys all hubs in that cluster

`python3 deploy.py deploy <cluster-name> <hub-name>` deploys just that one hub.

Maybe by adding cluster name & hub name as optional arguments to the deploy parser?

Reply:
Perfect! :+1: